vic/drivers/cesm/cpl_esmf/lnd_comp_esmf.F90

```diff
@@ -0,0 +1,178 @@
+
+    print *, "In lnd register routine"
+    ! Register the callback routines.
+
+    call ESMF_GridCompSetEntryPoint(comp, ESMF_METHOD_INITIALIZE, &
+      lnd_init_esmf, phase=1, rc=rc)
+    if (rc /= ESMF_SUCCESS) call ESMF_Finalize(rc=rc, endflag=ESMF_END_ABORT)
+    call ESMF_GridCompSetEntryPoint(comp, ESMF_METHOD_RUN, &
+      lnd_run_esmf, phase=1, rc=rc)
+    if (rc /= ESMF_SUCCESS) call ESMF_Finalize(rc=rc, endflag=ESMF_END_ABORT)
+    call ESMF_GridCompSetEntryPoint(comp, ESMF_METHOD_FINALIZE, &
+      lnd_final_esmf, phase=1, rc=rc)
+    if (rc /= ESMF_SUCCESS) call ESMF_Finalize(rc=rc, endflag=ESMF_END_ABORT)
+
+
+end subroutine
+
+!===============================================================================
+
+!===============================================================================
+!BOP ===========================================================================
+!
+! !IROUTINE: lnd_init_esmf
+!
+! !DESCRIPTION:
+!     initialize the VIC land model
+!
+! !REVISION HISTORY:
+!
+! !INTERFACE: ------------------------------------------------------------------
+
+subroutine lnd_init_esmf(comp, import_state, export_state, EClock, rc)
+
+! !INPUT/OUTPUT PARAMETERS:
+   type(ESMF_GridComp)          :: comp
+   type(ESMF_State)             :: import_state
+   type(ESMF_State)             :: export_state
+   type(ESMF_Clock)             :: EClock
+   integer, intent(out)         :: rc
+
+   ! Local variables
+   character(ESMF_MAXSTR) :: convCIM, purpComp
+
+!EOP
+
+    rc = ESMF_SUCCESS
+
+    ! Set flag to specify dead components
+    call ESMF_AttributeSet(export_state, name="lnd_present", value=.false., rc=rc)
+    if (rc /= ESMF_SUCCESS) call ESMF_Finalize(rc=rc, endflag=ESMF_END_ABORT)
+
+    call ESMF_AttributeSet(export_state, name="lnd_prognostic", value=.false., rc=rc)
+    if (rc /= ESMF_SUCCESS) call ESMF_Finalize(rc=rc, endflag=ESMF_END_ABORT)
+
+    call ESMF_AttributeSet(export_state, name="rof_present", value=.false., rc=rc)
+    if (rc /= ESMF_SUCCESS) call ESMF_Finalize(rc=rc, endflag=ESMF_END_ABORT)
+
+    call ESMF_AttributeSet(export_state, name="sno_present", value=.false., rc=rc)
+    if (rc /= ESMF_SUCCESS) call ESMF_Finalize(rc=rc, endflag=ESMF_END_ABORT)
+
+    call ESMF_AttributeSet(export_state, name="sno_prognostic", value=.false., rc=rc)
+    if (rc /= ESMF_SUCCESS) call ESMF_Finalize(rc=rc, endflag=ESMF_END_ABORT)
+
+    convCIM  = "CIM"
+    purpComp = "Model Component Simulation Description"
+
+    call ESMF_AttributeAdd(comp,  &
+                           convention=convCIM, purpose=purpComp, rc=rc)
+
+    call ESMF_AttributeSet(comp, "ShortName", "VIC", &
+                           convention=convCIM, purpose=purpComp, rc=rc)
+    call ESMF_AttributeSet(comp, "LongName", &
+                           "Variable Infiltration Capacity Model", &
+                           convention=convCIM, purpose=purpComp, rc=rc)
+    call ESMF_AttributeSet(comp, "ReleaseDate", "2014", &
```

Comment:
Do we want this to be the year of the VIC release. In that case, it should be 2016